crates/fiber/src/arch/aarch64.S

```diff
@@ -1,194 +0,0 @@
-// A WORD OF CAUTION
-//
-// This entire file basically needs to be kept in sync with itself. It's not
-// really possible to modify just one bit of this file without understanding
-// all the other bits. Documentation tries to reference various bits here and
-// there but try to make sure to read over everything before tweaking things!
-//
-// Also at this time this file is heavily based off the x86_64 file, so you'll
-// probably want to read that one as well.
-//
-// Finally, control flow integrity hardening has been applied to the code using
-// the Pointer Authentication (PAuth) and Branch Target Identification (BTI)
-// technologies from the Arm instruction set architecture:
-// * All callable functions start with either the `BTI c` or `PACIASP`/`PACIBSP`
-//   instructions
-// * Return addresses are signed and authenticated using the stack pointer
-//   value as a modifier (similarly to the salt in a HMAC operation); the
-//   `DW_CFA_AARCH64_negate_ra_state` DWARF operation (aliased with the
-//   `.cfi_window_save` assembler directive) informs an unwinder about this
-
-#include "header.h"
-
-#ifndef CFG_TARGET_OS_macos
-// We need to tell whatever loads the following code (e.g. the dynamic linker)
-// that it is compatible with BTI, so that the corresponding executable memory
-// pages have the necessary attribute set (if supported by the environment). To
-// this end, we follow the ELF for the Arm® 64-bit Architecture standard, and
-// use a special metadata section. Further details are in section 6.2 of the
-// specification:
-//
-// https://github.com/ARM-software/abi-aa/blob/2022Q1/aaelf64/aaelf64.rst#program-property
-//
-// We also set the PAuth (PAC) property, even though it is optional, for the
-// sake of completeness.
-.pushsection .note.gnu.property, "a";
-.p2align 3;
-.word 4;
-.word 16;
-.word 5;
-.asciz "GNU";
-.word 0xc0000000; // GNU_PROPERTY_AARCH64_FEATURE_1_AND
-.word 4;
-.word 3; // GNU_PROPERTY_AARCH64_FEATURE_1_BTI | GNU_PROPERTY_AARCH64_FEATURE_1_PAC
-.word 0;
-.popsection
-#endif
```

Comment:
@akirilov-arm I have temporarily dropped this block in the hopes that it's only required for external assembly files rather than `global_asm!` blocks in Rust itself. I am not super familiar with `global_asm!` though so I could be wrong here. Could you help by taking a peek at the artifacts of this PR (when they're available) and see if everything looks correct?

Reply:
For the sake of other people who might look at this PR - we are also [discussing](https://bytecodealliance.zulipchat.com/#narrow/stream/217126-wasmtime/topic/verify.20BTI.20bits.20in.20binary) this issue in the Zulip chat.

Reply:
Ok while very roundabout I have confirmed that I don't believe it's necessary to include this section in the `global_asm!` anywhere. The `global_asm!` gets shoved directly into the final object file and the object files coming out of rustc otherwise have this section already configured with the `-Zbranch-protection=bti` flag. 

As I mentioned in Zulip there's a whole mess of other reason why the note won't show up but they're all unrelated to this stanza in this assembly (the other issues being that (a) your local rust precompiled standard library needs the note and (b) your local glibc toolchain with `crt*.o` also needs the note, neither of which have it by default)